### PR TITLE
Analyzer MVP

### PR DESCRIPTION
**Pull request by DerekStride**
@ismail-syed @justinkrol 
Some initial work on the analyzer.

The format for the output comes from the [extensions response format](http://facebook.github.io/graphql/October2016/#sec-Response-Format) from the GraphQL Spec and the [apollo tracing format](https://github.com/apollographql/apollo-tracing).

Overall the `Graphql::Analyzer` and `Graphql::Analyzer::Instrumentation` classes and specs are the interesting parts of this PR.

To 🎩 
1. start the rails server `bundle exec rails s`
1. visit `http://localhost:3000/graphiql-analyze`
1. submit a query and look for the `extensions` key in the response.

### CODE REVIEW

File: spec/models/graphql/analyzer_spec.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
+require 'rails_helper'
+
+describe Graphql::Analyzer do
+  let(:user) { create(:user) }
+  let(:query_string) { %|{ user(id: "#{user.to_global_id}") { id } }| }
+  let(:analyzer) { Graphql::Analyzer.new(AppSchema) }
```
Comment (DerekStride) [inline spec/models/graphql/analyzer_spec.rb]: Here is where we "wrap" the schema with a decorator that will add functionality that will return data relating to the queries that were made.

File: app/models/graphql/analyzer.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module Graphql
+  class Analyzer
+    def initialize(graphql_schema)
+      @graphql_schema = graphql_schema
+    end
+
+    def execute(*args)
+      instrumenter = Instrumentation.new
+      result = schema(instrumenter).execute(*args)
+      result['instrumentation'] = instrumenter
```
Comment (DerekStride) [inline app/models/graphql/analyzer.rb]: This should be `extensions` as per the [GraphQL spec](http://facebook.github.io/graphql/October2016/#sec-Response-Format)

File: spec/models/utils/misc_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
+def path(field_names)
```
Comment (DerekStride) [inline spec/models/utils/misc_spec.rb]: I'm going to 🔥 this whole file.

File: app/graph/tracers/stdout_tracer.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+class StdoutTracer
+  def self.trace(key, data)
```
Comment (DerekStride) [inline app/graph/tracers/stdout_tracer.rb:2]: I've added this so we can use it for some debugging, it has been quite useful to me but isn't part of this PR.

**Review by ismail-syed — APPROVED** [review]
This is looking fantastic! 🎩 went well. 
This design seems to be coupled with webscale. Will this be easy to pull out into the `graphql-ruby` gem?

**Pull request merged**